File: python/messages/controll_message.py
import json

from .message import Message


class ControllMessage(Message):
    """
    Message specifying action to perform for a step
    """
    def __init__(self, moveInput, turnInput, envNum):
        self.moveInput = float(moveInput)  # convert to float to make sure its serializable
        self.turnInput = float(turnInput)  # (np.float32 are not)
        self.envNum = envNum

File: Unity/SimpleDemoRL/Assets/Scripts/client/api/messages/ControllMessage.cs
using UnityEngine;

[System.Serializable]

public class AgentAction
{
    /*
    Message specifying action to perform for a step
    */
    public float moveInput;
    public float turnInput;
    public int envNum;


    public static AgentAction FromJson(string json)
    {
        return JsonUtility.FromJson<AgentAction>(json);
    }

    public string ToJson()
    {
        return JsonUtility.ToJson(this);
    }
}
